soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -1191,81 +1191,542 @@ s16 Randomizer::GetRandomizedItemId(GetItemID ogId, s16 actorId, s16 actorParams
+        case RG_MIRROR_SHIELD:
+            return !CHECK_OWNED_EQUIP(EQUIP_SHIELD, 2) ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_GORON_TUNIC:
+        case RG_BUY_GORON_TUNIC:
+            return !CHECK_OWNED_EQUIP(EQUIP_TUNIC, 1) ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_ZORA_TUNIC:
+        case RG_BUY_ZORA_TUNIC:
+            return !CHECK_OWNED_EQUIP(EQUIP_TUNIC, 2) ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_IRON_BOOTS:
+            return !CHECK_OWNED_EQUIP(EQUIP_BOOTS, 1) ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_HOVER_BOOTS:
+            return !CHECK_OWNED_EQUIP(EQUIP_BOOTS, 2) ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+
+        // Inventory Items
+        case RG_PROGRESSIVE_STICK_UPGRADE:
+            return CUR_UPG_VALUE(UPG_STICKS) < 3 ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_PROGRESSIVE_NUT_UPGRADE:
+            return CUR_UPG_VALUE(UPG_NUTS) < 3 ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_PROGRESSIVE_BOMB_BAG:
+            return CUR_UPG_VALUE(UPG_BOMB_BAG) < 3 ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_BOMBS_5:
+        case RG_BOMBS_10:
+        case RG_BOMBS_20:
+        case RG_BUY_BOMBS_525:
+        case RG_BUY_BOMBS_535:
+        case RG_BUY_BOMBS_10:
+        case RG_BUY_BOMBS_20:
+        case RG_BUY_BOMBS_30:
+            return CUR_UPG_VALUE(UPG_BOMB_BAG) ? CAN_OBTAIN : CANT_OBTAIN_NEED_UPGRADE;
+        case RG_PROGRESSIVE_BOW:
+            return CUR_UPG_VALUE(UPG_QUIVER) < 3 ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_ARROWS_5:
+        case RG_ARROWS_10:
+        case RG_ARROWS_30:
+        case RG_BUY_ARROWS_10:
+        case RG_BUY_ARROWS_30:
+        case RG_BUY_ARROWS_50:
+            return CUR_UPG_VALUE(UPG_QUIVER) ? CAN_OBTAIN : CANT_OBTAIN_NEED_UPGRADE;
+        case RG_PROGRESSIVE_SLINGSHOT:
+            return CUR_UPG_VALUE(UPG_BULLET_BAG) < 3 ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_DEKU_SEEDS_30:
+        case RG_BUY_DEKU_SEEDS_30:
+            return CUR_UPG_VALUE(UPG_BULLET_BAG) ? CAN_OBTAIN : CANT_OBTAIN_NEED_UPGRADE;
+        case RG_PROGRESSIVE_OCARINA:
+            switch (INV_CONTENT(ITEM_OCARINA_FAIRY)) {
+                case ITEM_NONE:
+                case ITEM_OCARINA_FAIRY:
+                    return CAN_OBTAIN;
+                case ITEM_OCARINA_TIME:
+                default:
+                    return CANT_OBTAIN_ALREADY_HAVE;
+            }
+        case RG_PROGRESSIVE_BOMBCHUS:
+            return INV_CONTENT(ITEM_BOMBCHU) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_BOMBCHU_5:
+        case RG_BOMBCHU_10:
+        case RG_BOMBCHU_20:
+        case RG_BOMBCHU_DROP:
+        case RG_BUY_BOMBCHU_10:
+        case RG_BUY_BOMBCHU_20:
+        case RG_BUY_BOMBCHU_5:
+            return INV_CONTENT(ITEM_BOMBCHU) == ITEM_BOMBCHU ? CAN_OBTAIN : CANT_OBTAIN_NEED_UPGRADE;
```

Comment:
This may need an extra case added to it for Bombchus in logic. Just FYI. Bombchus in Logic allows for buying Bombchus without having already obtained them elsewhere. Unless that's what `RG_PROGRESSIVE_BOMBCHUS` is, in which case ignore me.

Reply:
Nope, good call.

Reply:
Although, I might have to leave a TODO for it, doesn't look like bombchu's in logic even has an RSK value yet

Reply:
Oh yeah for sure I was just giving you a heads up.